app/jobs/scraper/export_file_attachments_job.rb

```diff
@@ -0,0 +1,67 @@
+class Scraper::ExportFileAttachmentsJob < ApplicationJob
+  queue_as :default
+
+  TEMP_FILE_PATH = "#{Rails.root}/tmp/foo.zip"
+
+  def perform(source_file)
+    zip_file = create_zip_version_of_source_file(source_file)
+    file_names = unzip_attachments_and_list_file_names(zip_file)
+    save_attachments_to_db(file_names, source_file)
+    delete_extracted_files(file_names, zip_file)
+  end
+
+  private
+
+  def create_zip_version_of_source_file(source_file)
+    zip_file = Tempfile.open(TEMP_FILE_PATH, encoding: "ascii-8bit")
+    source_file.active_storage_attachment.blob.download { |chunk| zip_file.write(chunk) }
+    zip_file
+  end
+
+  def unzip_attachments_and_list_file_names(zip_file)
+    file_names = []
+
+    if !File.zero?(zip_file)
+      Zip::File.open(zip_file) do |zip_file|
+        zip_file.each do |entry|
+          next unless entry.name.include?("docx") || entry.name.include?(".bin")
```

Comment:
We probably want to make sure that both docx and bin are present in the filename as the extension and not just in the middle (I would be surprise if that's the case, but who knows). We can use `end_with?` to guarantee this

Reply:
Ooo nice!